File: ts/src/emitter/mermaid/theme/light-theme.test.ts
import { describe, expect, test } from "vitest";

import { lightTheme } from "./light-theme.js";

describe("lightTheme", () => {
  test("every classDef slot is populated", () => {
    expect(lightTheme.boundaryStub.stroke).not.toEqual("");
    expect(lightTheme.boundaryStub.strokeDasharray).not.toEqual("");
    expect(lightTheme.boundaryStub.color).not.toEqual("");
    expect(lightTheme.varNode.strokeDasharray).not.toEqual("");
    expect(lightTheme.elkEmptyPlaceholder.fill).not.toEqual("");
    expect(lightTheme.elkEmptyPlaceholder.stroke).not.toEqual("");
  });

  test("nestPalette has at least six entries to keep wrap / body brightness distinct", () => {
    expect(lightTheme.nestPalette.length >= 6).toEqual(true);
  });

  // The placeholder is a layout-only hack -- it is not a node. fill and
  // stroke stay transparent so no rectangle is drawn around the label.
  // Text color is left to Mermaid's default so the "No nodes" label
  // stays readable against whichever subgraph background it lands on.
  test("elkEmptyPlaceholder has transparent fill and stroke (same workaround as the dark theme)", () => {
    expect(lightTheme.elkEmptyPlaceholder.fill).toEqual("transparent");
    expect(lightTheme.elkEmptyPlaceholder.stroke).toEqual("transparent");
  });
});

File: crates/unsnarl-emitter-mermaid/src/theme/light_theme_test.rs
//! Mirrors `ts/src/emitter/mermaid/theme/light-theme.test.ts`. Pins
//! that the light theme has every slot populated and shares the
//! same transparent elkEmptyPlaceholder workaround as the dark theme.

use super::LIGHT_THEME;

#[test]
fn every_class_def_slot_is_populated() {
    assert!(!LIGHT_THEME.boundary_stub.stroke.is_empty());
    assert!(!LIGHT_THEME.boundary_stub.stroke_dasharray.is_empty());
    assert!(!LIGHT_THEME.boundary_stub.color.is_empty());
    assert!(!LIGHT_THEME.var_node.stroke_dasharray.is_empty());
    assert!(!LIGHT_THEME.elk_empty_placeholder.fill.is_empty());
    assert!(!LIGHT_THEME.elk_empty_placeholder.stroke.is_empty());
}

#[test]
fn nest_palette_has_at_least_six_entries_to_keep_wrap_body_brightness_distinct() {
    assert!(LIGHT_THEME.nest_palette.len() >= 6);
}

#[test]
fn elk_empty_placeholder_has_transparent_fill_and_stroke_same_workaround_as_dark_theme() {
    // The placeholder is a layout-only hack — it is not a node. fill
    // and stroke stay transparent so no rectangle is drawn around
    // the label. Text color is left to Mermaid's default so the
    // "No nodes" label stays readable against whichever subgraph
    // background it lands on.
    assert_eq!(LIGHT_THEME.elk_empty_placeholder.fill, "transparent");
    assert_eq!(LIGHT_THEME.elk_empty_placeholder.stroke, "transparent");
}
